subsys/pm/power.c

```diff
@@ -226,14 +229,20 @@ void pm_system_resume(void)
 	 * notification is not required.
 	 */
 	if (!post_ops_done) {
-		post_ops_done = 1;
-		exit_pos_ops(z_power_state);
+		uint8_t id = _current_cpu->id;
+
+		post_ops_done = true;
+		exit_pos_ops(z_power_states[id]);
 		pm_state_notify(false);
+		z_power_states[id] = (struct pm_state_info){PM_STATE_ACTIVE,
+			0, 0};
 	}
 }
 
 void pm_power_state_force(struct pm_state_info info)
```

Comment:
Maybe I'm wrong, but I feel as if this should take the cpu id as a parameter to force the state on here, otherwise the API and docs make it seems like the call would be a system wide power state and not for the currently cpu context

There may be cases as well where a task on one cpu decides when to force the other to a particular state.

I get this has some maybe unwanted implications, but it does feel like the better API to have. Maybe that needs to be built on top of this though, so that you schedule a task to run this function on the cpu you want to force a power state on.

Reply:
I will remove this function. Will do it in a different PR to bring more visibility for this. The force behind this is that this API is very problematic in SMP environments. For example, one cpu may suspend another one that was in holding a mutex or spin lock that will cause a problem to the system. In this situation what should be done ? abort the thread running in that core ? just scheduled out the thread ? but if it is pinned to the suspended core ?